hw/ip/pattgen/data/pattgen.hjson

```diff
@@ -0,0 +1,160 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+{ name: "pattgen",
+  clock_primary: "clk_i",
+  bus_device: "tlul",
+  bus_host: "none",
+  estimated_gate: "1k",
+  regwidth: "32",
+  // OUTPUT pins
+  available_output_list: [
+    { name: "pda0_tx", desc: "Serial output data bit for pattern generation on channel 0" }
+    { name: "pcl0_tx", desc: "Bit-banging output clock for pattern generation on channel 0" }
+    { name: "pda1_tx", desc: "Serial output data bit for pattern generation on channel 1" }
+    { name: "pcl1_tx", desc: "Bit-banging output clock for pattern generation on channel 1" }
+  ]
+  // INTERRUPT pins
+  interrupt_list: [
+    { name: "patt_done0"
+      desc: "raise if pattern generation on channel 0 is complete"
+    }
+    { name: "patt_done1"
+      desc: "raise if pattern generation on channel 1 is complete"
+    }
+  ],
+  param_list: [
+    { name:    "NumRegsData",
+      type:    "int",
+      default: "2",
+      desc:    "Number of data registers per each channel",
+      local:   "true"
+    }
+  ],
+  // REGISTER definition
+  registers: [
+    // CTRL register
+    { name: "PATT_CTRL",
+      desc: "PATTGEN control register",
+      swaccess: "rw",
+      hwaccess: "hro",
+      fields: [
+        { bits: "0",
+          resval: "0",
+          name: "ENABLE_CH0",
+          desc: "Enable pattern generator functionality for channel 0"
+        }
+        { bits: "1",
+          resval: "0",
+          name: "ENABLE_CH1",
+          desc: "Enable pattern generator functionality for channel 1"
+        }
+        { bits: "2",
+          resval: "0",
+          name: "POLARITY_CH0",
+          desc: "Clock (`pcl`) polarity for channel 0.  If high, `pda` signal changes on falling edge of pcl signal, otherwise pda changes on rising edge"
+        }
+        { bits: "3",
+          resval: "0",
+          name: "POLARITY_CH1",
+	  desc: "Clock (`pcl`) polarity for channel 1.  If high, `pda` signal changes on falling edge of `pcl` signal, otherwise pda changes on rising edge"
```

Comment:
Please indent with spaces only.

Reply:
Ugh.  Any advice on how to quickly see if I've included tabs or spaces?  I'm happy to comply, but aside from running my code through shell scripts.  I can't even tell the difference, nor can I see how you've identified this.

Reply:
BTW (unrelated to Markdown, I admit) but did I hear something about a verible-based RTL formatter? Is there a note on how to use that for OT?

Reply:
Thanks for the comment Philipp.  After whining ignominiously (and publicly ;) ), I've learned how to use emacs whitespace-mode.   Though I am still curious where you see these tabs!